auth/src/test/java/io/camunda/zeebe/auth/JwtAuthorizationTest.java

```diff
@@ -0,0 +1,151 @@
+        claims.get(JwtAuthorizationEncoder.AUTHORIZED_TENANTS_CLAIM).as(List.class);
+    assertThat(authorizedTenantClaim).containsExactlyElementsOf(authorizedTenants);
+  }
+
+  @Test
+  public void shouldValidateAndDecodeJwtTokenWithDefaultClaims() {
+    // given
+    final String jwtToken =
+        JWT.create()
+            .withIssuer(defaultIssuer)
+            .withAudience(defaultAudience)
+            .withSubject(defaultSubject)
+            .sign(Algorithm.none());
+
+    // when
+    final JwtAuthorizationDecoder decoder = Authorization.decodeWithJwt().withJwtToken(jwtToken);
+    final Map<String, Claim> claims = decoder.getAuthorizations();
+
+    // then
+    assertDefaultClaims(claims);
+  }
+
+  @Test
+  public void shouldValidateAndDecodeJwtTokenWithAuthorizedTenantsClaim() {
+    // given
+    final List<String> authorizedTenants = List.of("tenant-1", "tenant-2", "tenant-3");
+    final String jwtToken =
+        JWT.create()
+            .withIssuer(defaultIssuer)
+            .withAudience(defaultAudience)
+            .withSubject(defaultSubject)
+            .withClaim(JwtAuthorizationEncoder.AUTHORIZED_TENANTS_CLAIM, authorizedTenants)
+            .sign(Algorithm.none());
+
+    // when
+    final JwtAuthorizationDecoder decoder =
+        Authorization.decodeWithJwt()
+            .withClaim(JwtAuthorizationEncoder.AUTHORIZED_TENANTS_CLAIM)
+            .withJwtToken(jwtToken);
+    final Map<String, Claim> claims = decoder.getAuthorizations();
+
+    // then
+    assertDefaultClaims(claims);
+    final List<String> authorizedTenantClaim =
+        claims.get(JwtAuthorizationEncoder.AUTHORIZED_TENANTS_CLAIM).as(List.class);
+    assertThat(authorizedTenantClaim).containsExactlyElementsOf(authorizedTenants);
+  }
+
+  @Test
+  public void shouldFailJwtTokenValidationWithNoAuthorizedTenants() {
+    // given
+    final String jwtToken =
+        JWT.create()
+            .withIssuer(defaultIssuer)
+            .withAudience(defaultAudience)
+            .withSubject(defaultSubject)
+            .sign(Algorithm.none());
+
+    // when
+    final JwtAuthorizationDecoder decoder =
+        Authorization.decodeWithJwt()
+            .withClaim(JwtAuthorizationEncoder.AUTHORIZED_TENANTS_CLAIM)
+            .withJwtToken(jwtToken);
+
+    // then
+    assertThatThrownBy(() -> decoder.getAuthorizations())
+        .isInstanceOf(UnrecoverableException.class)
+        .hasMessage(
+            "Authorization data unavailable: The Claim 'authorized_tenants' is not present in the JWT.");
+  }
+
+  @Test
+  public void shouldFailJwtTokenDecodingWithoutJwtToken() {
+    // when
+    final JwtAuthorizationDecoder decoder =
+        Authorization.decodeWithJwt().withClaim(JwtAuthorizationEncoder.AUTHORIZED_TENANTS_CLAIM);
+
+    // then
+    assertThatThrownBy(() -> decoder.getAuthorizations())
+        .isInstanceOf(UnrecoverableException.class)
+        .hasMessage("Authorization data unavailable: JWT token");
+  }
```

Comment:
🔧 I wonder if it makes more sense to pass the JWT as a constructor parameter to the decoder. If we do that we won't ever not have a JWT token.

Reply:
Sure, it might have a better UX. 

It would still be possible to pass an empty String as the JWT token. But then the `JWTVerifier` will throw an exception that the token is `null`. So we don't have to duplicate the validation.